pkg/build/api/v1beta1/types.go

```diff
@@ -111,6 +111,10 @@ type BuildSource struct {
 	// This allows to have buildable sources in directory other than root of
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
+
+	// SourceSecretName is the name of a Secret that would be used for setting
+	// up the authentication for cloning private repository.
```

Comment:
I don't think I should explain how to setup a secret over here, it'll only pollute build docs with unnecessary details,  which should be covered in secrets. Of course there will be some more information around that topic in our docs, but I don't think it's necessary here.